packages/vuetify/src/composables/filter.ts

```diff
@@ -178,8 +179,19 @@ export function useFilter <T extends InternalItem> (
       _filteredItems.push(item)
       _filteredMatches.set(item.value, matches)
     })
-    filteredItems.value = _filteredItems
-    filteredMatches.value = _filteredMatches
+
+    if (options?.forRegisteredNodes) {
+      // list items must be unmounted to reliably unregister
+      filteredItems.value = []
+      filteredMatches.value = new Map()
```

Comment:
This shouldn't be necessary if the items are keyed.

Reply:
Without this block you will observe warnings upon typing `t`. I have added console logs to make it clear what happens.

![image](https://github.com/user-attachments/assets/c30234ba-3417-4809-b5e5-a1177804a61a)